.github/workflows/project_automation.yml

```diff
@@ -0,0 +1,141 @@
+name: Project automations
+on:
+  issues:
+    types: [opened, reopened, assigned]
+  pull_request_target:
+    types: [opened, reopened, ready_for_review, review_requested, converted_to_draft, closed]
+  pull_request_review:
+    types: [submitted]
+
+# map fields with customized labels
+env:
+  organization: nipoppy
+  project_id: 2
+  # project statuses
+  triage: Triage
+  todo: Todo
+  in_progress: In progress
+  pending_review: Pending review
+  reviewed: Reviewed
+  approved: Approved
+  done: Done
+  cancelled: Cancelled
+
+jobs:
+
+  issue_opened_or_reopened:
+    name: issue_opened_or_reopened
+    runs-on: ubuntu-latest
+    if: github.event_name == 'issues' && (github.event.action == 'opened' || github.event.action == 'reopened')
+    steps:
+      - name: Move issue to ${{ env.triage }}
+        uses: leonsteinhaeuser/project-beta-automations@v2.2.1
+        with:
+          gh_token: ${{ secrets.PROJECT_PAT }}
+          organization: ${{ env.organization }}
+          project_id: ${{ env.project_id }}
+          resource_node_id: ${{ github.event.issue.node_id }}
+          status_value: ${{ env.triage }} # Target status
+
+  issue_assigned:
+    name: issue_assigned
+    runs-on: ubuntu-latest
+    if: github.event_name == 'issues' && github.event.action == 'assigned'
+    steps:
+      - name: Move issue to ${{ env.in_progress }}
+        uses: leonsteinhaeuser/project-beta-automations@v2.2.1
+        with:
+          gh_token: ${{ secrets.PROJECT_PAT }}
+          organization: ${{ env.organization }}
+          project_id: ${{ env.project_id }}
+          resource_node_id: ${{ github.event.issue.node_id }}
+          status_value: ${{ env.in_progress }} # Target status
+
+  pr_converted_to_draft:
+    name: pr_converted_to_draft
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request_target' && github.event.action == 'converted_to_draft'
+    steps:
+      - name: Move PR to ${{ env.in_progress }}
+        uses: leonsteinhaeuser/project-beta-automations@v2.2.1
+        with:
+          gh_token: ${{ secrets.PROJECT_PAT }}
+          organization: ${{ env.organization }}
+          project_id: ${{ env.project_id }}
+          resource_node_id: ${{ github.event.pull_request.node_id }}
+          status_value: ${{ env.in_progress }} # Target status
+          move_related_issues: true
+
+  pr_pending_review:
+    name: pr_pending_review
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request_target' && github.event.pull_request.draft == false && (github.event.action == 'opened' || github.event.action == 'reopened' || github.event.action == 'ready_for_review' || github.event.action == 'review_requested')
```

Comment:
Not sure about this part `github.event.action == 'opened' || github.event.action == 'reopened'`. I feel like this will tag a lot of issues as pending review. For example, a lot of people may open PRs without drafting them.